src/accounts/biconomy/signerToBiconomySmartAccount.ts

```diff
@@ -0,0 +1,389 @@
+            args: [ecdsaModuleAddress, ecdsaOwnershipInitData, index]
+        }) as Hex
+    ])
+}
+
+const getAccountAddress = async <
+    TTransport extends Transport = Transport,
+    TChain extends Chain | undefined = Chain | undefined
+>({
+    client,
+    factoryAddress,
+    accountLogicAddress,
+    fallbackHandlerAddress,
+    ecdsaModuleAddress,
+    entryPoint,
+    owner,
+    index = 0n
+}: {
+    client: Client<TTransport, TChain>
+    factoryAddress: Address
+    accountLogicAddress: Address
+    fallbackHandlerAddress: Address
+    ecdsaModuleAddress: Address
+    owner: Address
+    entryPoint: Address
+    index?: bigint
+}): Promise<Address> => {
+    // @ts-ignore
+    const unusedClient = client // Ignoring the error for now
+    // @ts-ignore
+    const unusedEntryPoint = entryPoint // Ignoring the error for now
+    // 1.
+    /*const initCode = await getAccountInitCode({owner, index, factoryAddress, ecdsaModuleAddress})
+
+    return getSenderAddress(client, {
+        initCode,
+        entryPoint
+    })*/
+
+    //2.
+
+    // Build the module setup data
+    const ecdsaOwnershipInitData = encodeFunctionData({
+        abi: BiconomyInitAbi,
+        functionName: "initForSmartAccount",
+        args: [owner]
+    })
+
+    // Build account init code
+    const initialisationData = encodeFunctionData({
+        abi: BiconomyInitAbi,
+        functionName: "init",
+        args: [
+            fallbackHandlerAddress,
+            ecdsaModuleAddress,
+            ecdsaOwnershipInitData
+        ]
+    })
+
+    const deploymentCode = encodePacked(
+        ["bytes", "uint256"],
+        [BICONOMY_PROXY_CREATION_CODE, hexToBigInt(accountLogicAddress)]
+    )
+
+    const salt = keccak256(
+        encodePacked(
+            ["bytes32", "uint256"],
+            [keccak256(encodePacked(["bytes"], [initialisationData])), index]
+        )
+    )
+
+    return getContractAddress({
+        from: factoryAddress,
+        salt,
+        bytecode: deploymentCode,
+        opcode: "CREATE2"
+    })
+}
+
+/**
+ * Build a Biconomy modular smart account from a private key, that use the ECDSA signer behind the scene
+ * @param client
+ * @param privateKey
+ * @param entryPoint
+ * @param index
+ * @param factoryAddress
+ * @param accountLogicAddress
+ * @param ecdsaModuleAddress
+ */
+export async function signerToBiconomySmartAccount<
+    TTransport extends Transport = Transport,
+    TChain extends Chain | undefined = Chain | undefined
+>(
+    client: Client<TTransport, TChain>,
+    {
+        signer,
+        entryPoint,
+        index = 0n,
+        factoryAddress = BICONOMY_ADDRESSES.FACTORY_ADDRESS,
+        accountLogicAddress = BICONOMY_ADDRESSES.ACCOUNT_V2_0_LOGIC,
+        fallbackHandlerAddress = BICONOMY_ADDRESSES.DEFAULT_FALLBACK_HANDLER_ADDRESS,
+        ecdsaModuleAddress = BICONOMY_ADDRESSES.ECDSA_OWNERSHIP_REGISTRY_MODULE
+    }: {
+        signer: SmartAccountSigner
```

Comment:
`SmartAccountSigner` now needs source & address;

Reply:
can you share an example or link particular line/s in latest code

Reply:
ok I see

Reply:
fixed